### PR TITLE
feat: allow installing multiple envs at once

### PR DESCRIPTION
**Pull request by tdejager**
Allow installation of multiple envs at the same time. Eg:

```bash
pixi install -e default -e other
```

### CODE REVIEW

**Comment by Glatzel** [comment]
Can you also add an option like `pixi install --all` to allow installing all environments at once?
It will be useful when initializing a project.

**Comment by tdejager** [comment]
> Can you also add an option like `pixi install --all` to allow installing all environments at once? It will be useful when initializing a project.

Yes for sure! Was thinking I'll wait till someone asks for it, which was pretty soon 😄

**Comment by tdejager** [comment]
> Can you also add an option like `pixi install --all` to allow installing all environments at once? It will be useful when initializing a project.

Ok, added this!

**Comment by tdejager** [comment]
I've also added a message, what environments are installed, install is the only CLI command doing this currently. But when I was testing it out with multiple envs, it got a bit weird not having this. @ruben-arts @Glatzel wdyt?

**Comment by Glatzel** [comment]
> I've also added a message, what environments are installed, install is the only CLI command doing this currently. But when I was testing it out with multiple envs, it got a bit weird not having this. @ruben-arts @Glatzel wdyt?
> 
> Edit (eg):
> 
> ```shell
> > Installed environment: test!
> ✔ Project in /Users/tdejager/development/prefix/pixi/examples/flask-hello-world-pyproject is ready to use!
> ```

I got a `thread 'main' has overflowed its stack` in win pc. Maybe envs should be installed in order instead of parallel?

**Comment by tdejager** [comment]
> > I've also added a message, what environments are installed, install is the only CLI command doing this currently. But when I was testing it out with multiple envs, it got a bit weird not having this. @ruben-arts @Glatzel wdyt?
> > Edit (eg):
> > ```shell
> > > Installed environment: test!
> > ✔ Project in /Users/tdejager/development/prefix/pixi/examples/flask-hello-world-pyproject is ready to use!
> > ```
> 
> Hi, tdejager. I got a `thread 'main' has overflowed its stack` in win pc. Maybe envs should be installed in order instead of parallel?

That's weird, they are installed concurrently not in parallel. Could you share your `pixi.toml`?

**Comment by tdejager** [comment]
> > > I've also added a message, what environments are installed, install is the only CLI command doing this currently. But when I was testing it out with multiple envs, it got a bit weird not having this. @ruben-arts @Glatzel wdyt?
> > > Edit (eg):
> > > ```shell
> > > > Installed environment: test!
> > > ✔ Project in /Users/tdejager/development/prefix/pixi/examples/flask-hello-world-pyproject is ready to use!
> > > ```
> > 
> > 
> > Hi, tdejager. I got a `thread 'main' has overflowed its stack` in win pc. Maybe envs should be installed in order instead of parallel?
> 
> That's weird, they are installed concurrently not in parallel. Could you share your `pixi.toml`?

And does it also happen if you specify one or two environment e.g: `-e a -e b`.

**Comment by Glatzel** [comment]
Here is my pixi.toml.
Both `pixi install --all` and `pixi install -e py311` raise `thread 'main' has overflowed its stack`.
```
[project]
name = "Install-Multi_Envs"
platforms = ["win-64"]
channels = ["conda-forge"]

[dependencies]
python = ">=3.10"
[pypi-dependencies]
numpy = ">=1.23"
[feature.py311.dependencies]
python = "3.11.*"
[feature.py312.dependencies]
python = "3.12.*"

[environments]
py312 = ["py312"]
py311 = ["py311"]

```

File: src/cli/install.rs [hunk 1]
```diff
@@ -15,18 +16,52 @@ pub struct Args {
...
+        get_up_to_date_prefix(&environment, args.lock_file_usage.into(), false).await?;
+    }
+
+    let s = if installed_envs.len() > 1 { "s" } else { "" };
+    // Message what's installed
+    eprintln!(
+        "> Installed environment{s}: {}!",
+        installed_envs
+            .iter()
+            .map(|n| console::style(n).bold())
```
Comment (ruben-arts) [inline src/cli/install.rs:62]: The EnvironmentName has a `fancy_display` you can use to use a default style for all environments.

**Comment by Glatzel** [comment]
> I've also added a message, what environments are installed, install is the only CLI command doing this currently. But when I was testing it out with multiple envs, it got a bit weird not having this. @ruben-arts @Glatzel wdyt?
> 
> Edit (eg):
> 
> ```shell
> > Installed environment: test!
> ✔ Project in /Users/tdejager/development/prefix/pixi/examples/flask-hello-world-pyproject is ready to use!
> ```

I think this installed environments message is a good hint for me. Besides, maybe we can have a cli command for getting this message, like `pixi list --installed-environments`. But I am worried that users will confuse this command with the existed `pixi list --environments` command.

File: src/cli/install.rs [hunk 2]
```diff
@@ -15,18 +16,52 @@ pub struct Args {
...
+    for env in envs {
+        let environment = project.environment_from_name_or_env_var(Some(env))?;
+        installed_envs.push(environment.name().to_string());
+        get_up_to_date_prefix(&environment, args.lock_file_usage.into(), false).await?;
+    }
+
+    let s = if installed_envs.len() > 1 { "s" } else { "" };
+    // Message what's installed
+    eprintln!(
+        "> Installed environment{s}: {}!",
```
Comment (ruben-arts) [inline src/cli/install.rs:59]:
```suggestion
        "> Installed environment{s}: {}",
```

Another idea would be to remove the next print completely and use this to share that the environment is ready

```suggestion
        "> The following environment{s} are ready to use: \n\t{}",
```

File: src/cli/install.rs [hunk 3]
```diff
@@ -15,18 +16,52 @@ pub struct Args {
     pub lock_file_usage: super::LockFileUsageArgs,
 
     #[arg(long, short)]
-    pub environment: Option<String>,
+    pub environments: Option<Vec<String>>,
```
Comment (ruben-arts) [inline src/cli/install.rs:19]: I guess this should be renamed to `environment` as it doesn't act like multiple input is possible:
```
pixi install --environments schema docs
```

**Review by ruben-arts — COMMENTED** [review]
I like the changes, just a few points

File: src/cli/install.rs (hunk 3 above)
Comment (tdejager) [reply src/cli/install.rs:19]: Ah that's right, back to environment you mean :)?

File: src/cli/install.rs (hunk 2 above)
Comment (tdejager) [reply src/cli/install.rs:59]: Good idea, will change :)

File: src/cli/install.rs (hunk 1 above)
Comment (tdejager) [reply src/cli/install.rs:62]: Thanks, will change!

**Comment by tdejager** [comment]
@Glatzel does this happen with 'main' as well? I can't really imagine these changes causing this..🤔

**Comment by Glatzel** [comment]
> @Glatzel does this happen with 'main' as well? I can't really imagine these changes causing this..🤔

Well, I tried the `pixi install` command and raised the `thread 'main' has overflowed its stack` problem, seems there are some bugs in the current main branch.

Also tested in version 0.22.0, did not find this problem.

**Comment by tdejager** [comment]
> > @Glatzel does this happen with 'main' as well? I can't really imagine these changes causing this..🤔
> 
> 
> 
> Well, I tried the `pixi install` command and raised the `thread 'main' has overflowed its stack` problem, seems there are some bugs in the current main branch.
> 
> 
> 
> Also tested in version 0.22.0, did not find this problem.

Okay, so it will not block this PR but a serious bug nonetheless, could you make an issue? Thanks for helping with the initial debug!

**Comment by Glatzel** [comment]
> > > @Glatzel does this happen with 'main' as well? I can't really imagine these changes causing this..🤔
> > 
> > 
> > Well, I tried the `pixi install` command and raised the `thread 'main' has overflowed its stack` problem, seems there are some bugs in the current main branch.
> > Also tested in version 0.22.0, did not find this problem.
> 
> Okay, so it will not block this PR but a serious bug nonetheless, could you make an issue? Thanks for helping with the initial debug!

No problem. I have created this issue in #1414.

**Comment by ruben-arts** [comment]
Cleaned up the output:

image

File: src/cli/install.rs [hunk 4]
```diff
@@ -15,25 +16,61 @@ pub struct Args {
...
+    // Message what's installed
+    if installed_envs.len() == 1 {
+        eprintln!(
+            "{}The {} environment is installed.",
+            console::style(console::Emoji("✔ ", "")).green(),
+            installed_envs[0].fancy_display(),
+        );
+    } else {
+        eprintln!(
+            "{}The following environments where installed: \n\t{}",
```
Comment (olivier-lacroix) [inline src/cli/install.rs:65]:
```suggestion
            "{}The following environments have been installed: \n\t{}",
```

File: src/cli/install.rs [hunk 5]
```diff
@@ -15,25 +16,61 @@ pub struct Args {
...
+    for env in envs {
+        let environment = project.environment_from_name_or_env_var(Some(env))?;
+        get_up_to_date_prefix(&environment, args.lock_file_usage.into(), false).await?;
+        installed_envs.push(environment.name().clone());
+    }
+
+    // Message what's installed
+    if installed_envs.len() == 1 {
+        eprintln!(
+            "{}The {} environment is installed.",
```
Comment (olivier-lacroix) [inline src/cli/install.rs:59]:
```suggestion
            "{}The {} environment has been installed.",
```

**Review by ruben-arts — APPROVED** [review]

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**